File: sportbook_backend/models/booking.go
package models

import "gorm.io/gorm"

type Booking struct {
	gorm.Model
	UserID    uint   `json:"user_id"`
	User      User   `json:"user" gorm:"foreignKey:UserID"`
	FieldID   uint   `json:"field_id"`
	Field     Field  `json:"field" gorm:"foreignKey:FieldID"`
	Date      string `json:"date"`
	StartTime string `json:"start_time"`
	EndTime   string `json:"end_time"`
	Status    string `json:"status" gorm:"default:pending"`
	PaymentMethod string `json:"payment_method"`
	PaymentStatus string `json:"payment_status" gorm:"default:unpaid"`
	TotalPrice    int    `json:"total_price"`
}

File: sportbook_app/lib/models/booking.dart
class Booking {
  final int id;
  final int userId;
  final int fieldId;
  final String date;
  final String startTime;
  final String endTime;
  final String status;
  final String paymentMethod;
  final String paymentStatus;
  final int totalPrice;
  final Map<String, dynamic>? field;
  final Map<String, dynamic>? user;

  Booking({
    required this.id,
    required this.userId,
    required this.fieldId,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.status,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.totalPrice,
    this.field,
    this.user,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['ID'],
      userId: json['user_id'],
      fieldId: json['field_id'],
      date: json['date'],
      startTime: json['start_time'],
      endTime: json['end_time'],
      status: json['status'],
      paymentMethod: json['payment_method'] ?? 'cash',
      paymentStatus: json['payment_status'] ?? 'unpaid',
      totalPrice: json['total_price'] ?? 0,
      field: json['field'],
      user: json['user'],
    );
  }
}